uefi/src/proto/pci/mod.rs

```diff
@@ -54,12 +56,30 @@ impl PciIoAddress {
     }
 }
 
+impl From<u64> for PciIoAddress {
+    fn from(value: u64) -> Self {
+        unsafe { core::mem::transmute(value) }
```

Comment:
u64::from_le_bytes([self.reg,...])

might be a little more verbose but looks safer to me

Reply:
Added a unit-test for now.
Workin Option B:

```rust
impl From<u64> for PciIoAddress {
    fn from(value: u64) -> Self {
        let raw = value.to_ne_bytes();
        Self {
            reg: raw[0],
            fun: raw[1],
            dev: raw[2],
            bus: raw[3],
            ext_reg: u32::from_ne_bytes([raw[4], raw[5], raw[6], raw[7]])
        }
    }
}

impl From<PciIoAddress> for u64 {
    fn from(value: PciIoAddress) -> Self {
        let ereg = value.ext_reg.to_ne_bytes();
        u64::from_ne_bytes([value.reg, value.fun, value.dev, value.bus, ereg[0], ereg[1], ereg[2], ereg[3]])
    }
}
```

Reply:
Yep let's do option B, we should avoid unsafe unless it's absolutely needed.